### PR TITLE
Sprint16 mdd bcc probes

### PR DESCRIPTION
**Pull request by ncultra**
- filetop sensor converted to a python class with json input and output

### CODE REVIEW

File: tools/filetop.py [hunk 1]
```diff
@@ -20,191 +21,226 @@
...
-    if (S_ISREG(mode)) {
-        info.type = 'R';
-    } else if (S_ISSOCK(mode)) {
-        info.type = 'S';
-    } else {
-        info.type = 'O';
+    int trace_read_entry(struct pt_regs *ctx, struct file *file,
+        char __user *buf, size_t count)
+    {
+        return do_entry(ctx, file, buf, count, 1);
```
Comment (markfsanderson) [inline tools/filetop.py:105]: style: true/false instead of 1/0 ?

File: tools/filetop.py [hunk 2]
```diff
@@ -20,191 +21,226 @@
...
-            break
-    counts.clear()
-
-    countdown -= 1
-    if exiting or countdown == 0:
-        print("Detaching...")
-        exit()
+    """
+
+    def __init__(self, args, debug = 0):
```
Comment (markfsanderson) [inline tools/filetop.py:116]: style: True/False vs 1/0 ?

**Review by markfsanderson — APPROVED** [review]
Just some style remarks . .

File: tools/filetop.py (hunk 2 above)
Comment (ncultra) [reply tools/filetop.py:116]: good suggestion, I like a boolean type for clarity

**Pull request merged**